packages/superise/src/ckb/index.ts

```diff
@@ -0,0 +1,212 @@
+        cellOutput: { lock },
+      } = await input.getCell(this.client);
+
+      if (scripts.some(({ script }) => script.eq(lock))) {
+        continue;
+      }
+
+      if (lock.eq(addressObj.script)) {
+        scripts.push({
+          script: lock,
+          cellDeps: (
+            await this.client.getKnownScript(ccc.KnownScript.Secp256k1Blake160)
+          ).cellDeps,
+        });
+      } else if (
+        lock.codeHash === acp.codeHash &&
+        lock.hashType === acp.hashType &&
+        lock.args.startsWith(acp.args)
+      ) {
+        scripts.push({
+          script: lock,
+          cellDeps: (
+            await this.client.getKnownScript(ccc.KnownScript.AnyoneCanPay)
+          ).cellDeps,
+        });
+      }
+    }
+
+    return scripts;
+  }
+
+  public setUiMetadataForTx(
+    tx: ccc.TransactionLike,
+    metadata: SupeRISE.SignCkbHashAllMetadata,
+  ) {
+    const txHash = ccc.Transaction.from(tx).hash();
+    this._uiMetadataMap[txHash] = metadata;
+  }
+
+  private getUiMetadataFromTx(tx: ccc.TransactionLike) {
+    const txHash = ccc.Transaction.from(tx).hash();
+    const metadata = this._uiMetadataMap[txHash];
+    delete this._uiMetadataMap[txHash];
+    return metadata;
+  }
+
+  override async prepareTransaction(txLike: ccc.TransactionLike) {
+    const tx = ccc.Transaction.from(txLike);
+
+    const scripts = await this.getRelatedScripts(tx);
+
+    await Promise.all(
+      scripts.map(async ({ script, cellDeps }) => {
+        await tx.prepareSighashAllWitness(script, 65, this.client);
+        await tx.addCellDepInfos(this.client, cellDeps);
+      }),
+    );
+
+    return tx;
+  }
+
+  override async signOnlyTransaction(txLike: ccc.TransactionLike) {
+    const tx = ccc.Transaction.from(txLike);
+    const metadata = this.getUiMetadataFromTx(tx);
+
+    const signatureCache = new Map<string, string>();
+
+    for (const { script } of await this.getRelatedScripts(tx)) {
+      const info = await tx.getSignHashInfo(script, this.client);
+      if (!info) {
+        return tx;
+      }
+
+      const { message, position } = info;
+
+      let signature!: string;
+      if (signatureCache.has(message)) {
+        signature = signatureCache.get(message)!;
+      } else {
+        const sign = await this.bridge.signCkbHashAll(
+          message.replace(/^0x/, ""),
+          metadata,
+        );
+        signature = sign.signature;
+        signatureCache.set(message, signature);
+      }
+
+      const witness =
+        tx.getWitnessArgsAt(info.position) ?? ccc.WitnessArgs.from({});
+      witness.lock = signature as ccc.Hex;
+      tx.setWitnessArgsAt(position, witness);
+    }
+
+    return tx;
+  }
+
+  override async signTransaction(
+    tx: ccc.TransactionLike,
+  ): Promise<ccc.Transaction> {
+    const preparedTx = await this.prepareTransaction(tx);
+    return this.signOnlyTransaction(preparedTx);
+  }
+
+  override async sendTransaction(tx: ccc.TransactionLike): Promise<ccc.Hex> {
```

Comment:
It seems unnecessary to override `signTransaction` and `sendTransaction`, since they're identical to the default implementations.